src/idput.c

```diff
@@ -531,16 +531,33 @@ void id_put(struct board *src_board, unsigned char x_pos, unsigned char y_pos,
   draw_char_ext(c, color, x_pos, y_pos, 0, 0);
 }
 
-void draw_game_window(struct board *src_board, int array_x, int array_y)
+static void offset_layers(struct board *src_board, int scroll_x, int scroll_y)
+{
+  // Offset the board and overlay layers by the per-pixel viewport offsets
+  int offset_x = -(scroll_x % 8);
+  int offset_y = -(scroll_y % 14);
+  move_layer(BOARD_LAYER, offset_x, offset_y);
+  move_layer(OVERLAY_LAYER, offset_x, offset_y);
```

Comment:
Probably ought to have something like this here instead:
```
  if(!(src_board->overlay_mode & 2))
    move_layer(OVERLAY_LAYER, offset_x, offset_y);
```
This probably requires that more of the board be conditionally drawn separate from the overlay in `id_put()` though. Alternatively, could disable pixel scrolling altogether when a static overlay exists because this looks really annoying to work around.